src/python/WMCore/MicroService/DataStructs/Workflow.py

```diff
@@ -213,7 +214,16 @@ def setPrimaryBlocks(self, blocksDict):
         {"block_name": {"blockSize": 1234,
                         "locations": [list of locations]}}
         """
-        self.primaryBlocks = blocksDict
+        blockWhite = self.getBlockWhitelist()
+        if blockWhite:
+            for block in blockWhite:
+                if block in blocksDict:
+                    self.primaryBlocks[block] = blocksDict[block]
```

Comment:
Two lines below you do a deepcopy of blocksDict, while here you take the assingment. Since I don't know what blocksDict[block] data type is it may be dangerous in case if it is dict type.

Reply:
That's a good comment! blocksDict[block] is actually a flat dict. So indeed it's safer to do a `copy` here and a `deepcopy` below; even though I did some local tests and they didn't have the same id.

Reply:
@amaltaro , I would agree with Valentine's comment here. It is still unclear to me, why in that 'if' there are two different methods for copying. Why not, just to stay on the safe side, use deep copy in both cases? Especially since the data structure is not defined in the current function, but comes as an argument from this call [1] (I believe). It would be safer if something else, but a flat dict is passed to `setPrimaryBlocks`. And in the case when an actual flat dict is passed, it should still produce a legitimate copy. Otherwise it will be an easy error to make - changing the data structure itself, wherever it is defined, and forgetting to update how you copy here. 

[1]
https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/MicroService/Unified/RequestInfo.py#L310

Reply:
Thanks Todor and Valentin. I have changed that to a deepcopy in a PR that I have opened now. It's indeed a dangerous business when selective copying/removing things from such structures. 
We could make it cleaner by defining two different data structures, but that would cause a >25% increase in the memory usage.